kuksa_certificates/jwt/createToken.py

```diff
@@ -19,10 +19,8 @@ def createJWTToken(input_filename, priv_key):
 
     encoded = jwt.encode(payload, priv_key, algorithm="RS256")
 
-    if input_filename.endswith(".json"):
-        output_filename = input_filename[:-5] + ".token"
-    else:
-        output_filename = output_filename + ".token"
+    output_filename = input_filename[:-5] if input_filename.endswith(".json") else input_filename
```

Comment:
I have no objection to the bug correction itself, the code looks nice and great that you found and addressed it @landgraf.

But both the old and new implementation does not match the example tokens that we actually have in this repository. Like we have `single-read.json` and `single-read.json.token`, i.e. it seems that `.token` has been added and `.json` has not been removed. I sort of would prefer that this script matches the example tokens we have, i.e. either just add `.token` regardless of whether input is `single-read.json` or `single-read`, or change name of the example token. The latter would how ever require changes in all places currently using the tokens.

So maybe best to just have only

`output_filename = input_filename + ".token"`
  
i.e. do not care if input file ends with `.json` or not. The reason why I prefer this option rather than renaming the tokens is that the tokens in this directory are used by KUKSA Server (C++ only, i.e. not Databroker), and we currently do not want to spend too much effort on KUKSA.val Server

Reply:
@erikbosch 
I'm not sure If I got your comment.
Current code with suggested change works exactly like you've mentioned.

```
>>> input_filename = "single-read.jsob"
>>> input_filename = "single-read.json"
>>> output_filename = input_filename[:-5] if input_filename.endswith(".json") else input_filename
>>> output_filename += ".token"
>>> print(output_filename)
single-read.token

>>> input_filename = "single-read"
>>> output_filename = input_filename[:-5] if input_filename.endswith(".json") else input_filename
>>> output_filename += ".token"
>>> print(output_filename)
```

Reply:
I've modified code a bit to make it even simpler and clever. Please re-review

Reply:
This is not how current  code works.
Prior the change:

```
    if input_filename.endswith(".json"):
        output_filename = input_filename[:-5] + ".token"
```
If file ends with ".json" then ".json" suffix is dropped and output_filename will be ".token" (without json).
In other words suggested change doesn't change behavior of the positive branch but fixes negative (else) branch which was broken before..

```
    else:
        output_filename = output_filename + ".token"
```
^ This code is broken and fails with:
`NameError: name 'output_filename' is not defined
`
So either your example was not generated with the current version of the script or there's bug somewhere.

Reply:
Yes, I share your conclusion, the current example token pairs in the repository that all have the form `<something>.json` and `<something>.json.token` does not match how the script work. If you use  `<something>.json`  as input you get `<something>.token` as output. Which gives us three options:

- Keep the mismatch. Not optimal as I see it.
- Adapt the script so it matches the pattern used in the examples, i.e. never remove `.json`
- Rename the examples so they match how the scripts works. If this is done we would need to do a search/replace for all places using the current names.

Reply:
>  the current example token pairs in the repository that all have the form <something>.json and <something>.json.token does not match how the script work.

This is not correct.

All the access tokens in the `jwt` folder are named like this (created using this script).

Would it help (cause less confusion) to move this script to that folder instead? Because as you say, we are not supposed to put any effort in maintaining kuksa-val-server stuff..

Reply:
My bad, I only thought about the folder https://github.com/eclipse/kuksa.val/tree/master/kuksa_certificates/jwt, not that https://github.com/eclipse/kuksa.val/tree/master/jwt also use this script.

Yes, we could move the script, but if so we need update *.md files referencing the script. But I think that we anyhow would benefit from having a small `README.md` in this folder that explains how to regenerate the tokens if needed, which will be need if anyone want to use them after 2025-12-31.

As I propose a small README I can create a separate PR myself, as this PR itself does not change behavior for input files names `<something>.json` anyway.